utils/createMunicipalityList.tsx

```diff
@@ -90,15 +94,15 @@ export const rankData = (municipalities: Municipality[], selectedData: SelectedD
 export const listColumns = (
   selectedData: SelectedData,
   columnHeader: string,
+  t: TFunction,
 ): ColumnDef<RowData>[] => {
   const isClimatePlan = selectedData === 'Klimatplanerna'
   const isProcurement = selectedData === 'Upphandlingarna'
 
   const getFirstColumnHeader = () => {
-    let firstColumnHeader = 'Ranking'
-    firstColumnHeader = isClimatePlan ? 'Har plan?' : firstColumnHeader
-    firstColumnHeader = isProcurement ? 'Ställer krav?' : firstColumnHeader
-    return firstColumnHeader
+    if (isClimatePlan) return t('startPage:hasPlan')
+    if (isProcurement) return t('startPage:procurementDemands')
+    return t('startPage:ranking')
```

Comment:
Denna verkar inte fungera, vare sig Klimatplaner eller Upphandlingar får korrekt första kolumn utan istället den vanliga "Ranking" och siffror

Reply:
Även för den tredje kolumnen är trasig för både Upphandlingarna och Klimatplanerna och behöver ses över

Reply:
Oh no... It took me way too long to realize this bug was introduced when I separated the dataDescription keys from the actual translation strings used for labels. Because of the wider typescript type assuming `string` keys rather than keys of an union type like `'utslappen' | 'koldioxidbudgetarna'`, this was technically correct although obviously not.

Sidenote: when debugging this, it became clear that the dataset definitions have many responsibilities. Either they should have all rendering logic (like which field to display, and how to format it), or we separate this. But let's not do that now.

Reply:
I just found another potential issue that is introduced with this PR: `listColumns()` is called many more times than before. This is because we can't use the constant `dataDescriptions`, and instead need to get the translated version via `getDataDescriptions()`, which probably cause extra re-renders.

One good thing is that we do have some caching, which greatly speeds things up. But it is still significantly slower than before.

Reply:
I tested it and still see errors.

Regarding the climate plans, it looks good until you sort them. Then, position 14, which is Ockelbo (the municipality with the date furthest in the future) ends up on top. The municipalities that do not exceed their budgets end up at the bottom of the ranking but with the correct ranking number (1-13).

As for the chargers, the municipalities without chargers are ranked incorrectly. Instead of the municipality ranked 290 appearing first when sorting from the bottom up, the municipality ranked 286 appears, so the list goes from 286 to 290 and then <= 285.

<img width="854" alt="Screenshot 2024-04-23 at 11 44 23" src="https://github.com/Klimatbyran/klimatkollen/assets/7899748/9134dd0e-6725-4543-bfff-d4a7ca25016b">

<img width="854" alt="Screenshot 2024-04-23 at 11 44 08" src="https://github.com/Klimatbyran/klimatkollen/assets/7899748/d5e55440-f5ff-4270-8b53-e4fdf7e8c86b">

Reply:
Never mind above, error is also present in staging and prod